src/main/groovy/nebula/plugin/dependencylock/tasks/GenerateLockTask.groovy

```diff
@@ -141,28 +126,48 @@ class GenerateLockTask extends AbstractLockTask {
                     Configuration confThatWillNotBeLocked = subproject.configurations.findByName(nameToLock)
                     if (confThatWillNotBeLocked == null) {
                         String message = "Global lock warning: project '${subproject.name}' requested locking a configuration which cannot be locked: '${nameToLock}'"
-                        errorMessages.add(message)
+                        warnings.add(message)
+                    } else {
+                        warnings.addAll(provideWarningsForConfiguration(confThatWillNotBeLocked, subproject))
                     }
                 }
             }
 
-            if (errorMessages.size() > 0) {
-                errorMessages.add("Requested configurations for global locks must be resolvable, consumable, and without resolution alternatives.\n" +
+            if (warnings.size() > 0) {
+                warnings.add("Requested configurations for global locks must be resolvable, consumable, and without resolution alternatives.\n" +
```

Comment:
Do we have examples of projects locking consumable configurations? I would be interested in to hear more.

Reply:
As an example, a test that exercised this area of code had a test setup that requested:
```
dependencyLock.configurationNames = ['testRuntime', 'special']
```
and this produces the error seen below when the dependencies of each subproject are added to a [configuration on the parent project](https://github.com/nebula-plugins/gradle-dependency-lock-plugin/blob/9d5d584a757fe545c28d142c944667ba07aa2883/src/main/groovy/nebula/plugin/dependencylock/DependencyLockTaskConfigurer.groovy#L233-L255) to make a global lock. This ends up with this `testRuntime` configuration getting consumed and triggering the warning:

```
The testRuntime configuration has been deprecated for consumption. This will fail with an error in Gradle 7.0. Please use attributes to consume the testRuntimeElements configuration instead.
```